encrypt/ecies/ecies.go

```diff
@@ -0,0 +1,104 @@
+// Package ecies implements the Elliptic Curve Integrated Encryption Scheme (ECIES).
+package ecies
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"errors"
+	"hash"
+
+	"github.com/dedis/kyber"
+	"github.com/dedis/kyber/util/random"
+	"golang.org/x/crypto/hkdf"
+)
+
+// Encrypt first performs a DH key exchange, then HKDF-derives a shared secret
+// key (and nonce) and finally AEAD encrypts the given message using AES-GCM.
+// Encrypt returns the ephemeral elliptic curve point of the DH key exchange
+// and the ciphertext or an error.
+func Encrypt(group kyber.Group, hash func() hash.Hash, public kyber.Point, message []byte) ([]byte, []byte, error) {
```

Comment:
I'm wondering if the new hash function could maybe be at the end as an optional argument where it defaults to `sha256.New`. What do you think ?